include/thread_pool/mpmc_bounded_queue.hpp

```diff
@@ -180,13 +245,10 @@ inline bool MPMCBoundedQueue<T>::push(U&& data)
         intptr_t dif = (intptr_t)seq - (intptr_t)pos;
         if(dif == 0)
         {
-            if(m_enqueue_pos.compare_exchange_weak(
-                   pos, pos + 1, std::memory_order_relaxed))
-            {
+            if(m_enqueue_pos.compare_exchange_weak(pos, pos + 1, std::memory_order_relaxed))
```

Comment:
Use acquire_release semantics here for the same reason. Because of reordering there might be some writes to a wrong cell, even if intuitively it seems it can't be.

load a then do b based on value of a may become do b based on some predicted value of a then check if a is different undo b

If you use acquire_release semantics it is guaranteed not to happen

Reply:
Fixed in #3.